lib/dangermattic/plugins/common/reporter.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Danger
+  # Handles reporting messages in the context of Danger, allowing the generation of
+  # warnings, errors, and simple messages in a Danger run.
+  #
+  # @example Reporting a Warning
+  #   reporter.report(message: "This is a warning message", type: :warning)
+  #
+  # @example Reporting an Error
+  #   reporter.report(message: "This is an error message", type: :error)
+  #
+  # @see Automattic/dangermattic
+  # @tags tool, util, danger
+  #
+  class Reporter < Plugin
+    # Report a message to be posted by Danger as an error (failing the build), a warning or a simple message.
+    #
+    # @param message [String] The message to be reported to Danger.
+    # @param type [Symbol] The type of report. Possible values:
+    #                        - :warning (default): Reports a warning.
+    #                        - :error: Reports an error.
+    #                        - :message: Reports a simple message.
+    #                        - :none or nil: Takes no action.
+    #
+    # @return [void]
+    def report(message:, type: :warning)
+      return if message.nil? || message.empty? || type.nil?
+
```

Comment:
Another nipick. What do you think of trusting danger to know how to best handle `nil` and empty messages?

If so, we could simplify and remove this whole check, because if `type` is `nil` the `case` will simply not run into any of the `when` branches below.

Reply:
[Checked](https://github.com/danger/danger/blob/bd89c4d6f3c7650d766eb804000a10f0b2fb98ab/lib/danger/danger_core/plugins/dangerfile_messaging_plugin.rb#L126) just in case 🙂  But yeah, I think it can be worth simplifying. Changed on dfd264dd26db273b4a6f825f7d475f23b6063b4b